test/core/pipeline/test_pipeline.py

```diff
@@ -916,3 +917,69 @@ def test_pipeline_is_not_stuck_with_components_with_only_defaults():
     answers = res["answer_builder"]["answers"]
     assert len(answers) == 1
     assert answers[0].data == "Paris"
+
+
+def test_pipeline_is_not_stuck_with_components_with_only_defaults_as_first_components():
+    """
+    This tests verifies that a Pipeline doesn't get stuck running in a loop if
+    it has all the following characterics:
+    - The first Component has all defaults for its inputs
+    - The first Component receives one input from the user
+    - The first Component receives one input from a loop in the Pipeline
+    - The second Component has at least one default input
+    """
+
+    def fake_generator_run(self, prompt: str, generation_kwargs: Optional[Dict[str, Any]] = None):
+        # Simple hack to simulate a model returning a different reply after the
+        # the first time it's called
+        if getattr(fake_generator_run, "called", False):
+            return {"replies": ["Rome"]}
+        fake_generator_run.called = True
+        return {"replies": ["Paris"]}
```

Comment:
Why not `fake_generator = mock.MagicMock(side_effect=[{"replies": ["Rome"]}, {"replies": ["Paris"]}])`?

Reply:
I didn't think about it but it doesn't work either. I have no other easy way of setting a default value to `generation_kwargs` without defining a function. 

The `component_class` factory class doesn't support a way of doing that as of now. Ideally it should, I'll open an issue to track that.